gdal/doc/source/development/rfc/rfc76_ogrpythondrivers.rst

```diff
@@ -0,0 +1,394 @@
+.. _rfc-76:
+
+================================================================================
+RFC 76: OGR Python drivers
+================================================================================
+
+============== ============================
+Author:        Even Rouault
+Contact:       even.rouault @ spatialys.com
+Started:       2019-Nov-5
+Last updated:  2019-Nov-5
+Status:        Draft
+Target:        GDAL 3.1
+============== ============================
+
+Summary
+-------
+
+This RFC adds the capability to write OGR/vector drivers in Python.
+
+Motivation
+----------
+
+For some use cases that do not require lighting speed, or to deal with very
+niche formats (possibly in house format), it might be faster and more efficient
+to write a vector driver in Python rather than a GDAL C++ driver as currently required,
+or an ad-hoc converter.
+
+This could benefit to QGIS which currently has no way to use providers in
```

Comment:
Actually, you can create QGIS vector data provider in python: https://github.com/qgis/QGIS/blob/master/tests/src/python/test_provider_python.py

Reply:
good to know ! This probably didn't exist at the time I started this experiment, or I just didn't know :-)